pkgs/by-name/ch/chatzone-desktop/package.nix

```diff
@@ -0,0 +1,70 @@
+{ lib
+, appimageTools
+, runCommand
+, fetchurl
+, stdenvNoCC
+, makeDesktopItem
+, copyDesktopItems
+, makeWrapper
+}:
+
+let
+  pname = "chatzone-desktop";
+  version = "5.1.10";
+
+  src = fetchurl {
+    url = "https://cdn1.ozone.ru/s3/chatzone-clients/ci/20240129/254/chatzone-desktop-linux-5.1.10.AppImage";
+    hash = "sha256-8jmJyg+Y2n8jGpWYnBRFsnCPIzoggJxG7neF7g6QQwk=";
+  };
+
+  appimage = appimageTools.wrapType2 {
+    inherit pname version src;
+  };
+  appimageContents = appimageTools.extract {
+    inherit pname version src;
+  };
+  icon = runCommand "chatzone-app-icon.png" {} ''
+    cp ${appimageContents}/app_icon.png $out
+  '';
```

Comment:
If we copy the icon correctly, we also don't need those two and can inline appimage

Reply:
All done, thanks for the suggestion